src/tracking.F90

```diff
@@ -64,6 +74,18 @@ subroutine transport(p)
       call initialize_particle_track()
     endif
 
+    ! Recover particle coordinates for dd runs
+    if (dd_run .and. current_stage > 1) then
+      ! Move particle slightly forward in case particle is on a surface
+      xyz_temp = p % coord(1) % xyz
+      p % coord(1) % xyz = p % coord(1) % xyz + TINY_BIT * p % coord(1) % uvw
+      call find_cell(p, found_cell)
+      if (.not. found_cell) then
+        call fatal_error("Could not locate particle after crossing domain.")
+      end if
+      p % coord(1) % xyz = xyz_temp
+    end if
```

Comment:
Yeah, I tried to remove these lines and tested BEAVRS as I remembered the inconsistent re-locating occurs only at complex geometry when particle crosses domain after just moving to a surface (the coordinate is slightly inside the former cell). But it didn't appear this time (I scaled the population up to 100000). So I think you are right. The `find_cell` was updated to account for particle's direction so re-locating can be correctly handled inside the event loop. @smharper